src/encode.rs

```diff
@@ -46,9 +46,9 @@ pub fn encode_buffered(input: Slipmux) -> Vec<u8> {
     }
     let mut buffer: Vec<u8> = vec![];
     let length = match input {
-        Slipmux::Diagnostic(s) => {
-            buffer.resize(space_requirement(s.len()), 0);
-            encode(FrameType::Diagnostic, s.as_bytes(), &mut buffer)
+        Slipmux::Diagnostic(str) => {
+            buffer.resize(space_requirement(str.len()), 0);
+            encode(FrameType::Diagnostic, str.as_bytes(), &mut buffer)
```

Comment:
```suggestion
        Slipmux::Diagnostic(message) => {
            buffer.resize(space_requirement(message.len()), 0);
            encode(FrameType::Diagnostic, message.as_bytes(), &mut buffer)
```

I think `s` is fine here, but if it pleases you, I would suggest to use `message` instead. This is a) the terminology used in the [draft](https://datatracker.ietf.org/doc/html/draft-bormann-t2trg-slipmux-03#section-4) and b) `str` is also a datatype in Rust, which I find more confusing than `s` :p

Reply:
Other instances too. I do like the expansion in configurations from `s` to `config` though.

Reply:
`str` is as bad as `s` here as neither does explain something to the reader.